GliaWidgetsTests/SecureConversations/Coordinator/SecureConversations.CoordinatorTests.swift

```diff
@@ -0,0 +1,212 @@
+import Foundation
+import XCTest
+@testable import GliaWidgets
+
+final class SecureConversationsCoordinatorTests: XCTestCase {
+    var navigationPresenter = NavigationPresenter(with: NavigationController())
+    var coordinator: SecureConversations.Coordinator!
+
+    override func setUp() {
+        coordinator = createCoordinator()
+    }
+
+    func createCoordinator(
+        initialScreen: SecureConversations.InitialScreen = .welcome
+    ) -> SecureConversations.Coordinator {
+        return SecureConversations.Coordinator(
+            messagingInitialScreen: initialScreen,
+            viewFactory: .mock(),
+            navigationPresenter: navigationPresenter,
+            environment: .mock
+        )
+    }
+
+    // Start
+    func test_startGeneratesWelcomeViewController() {
+        let viewController = coordinator.start() as? SecureConversations.WelcomeViewController
+
+        XCTAssertNotNil(viewController)
+    }
+
+    func test_startGeneratesChatViewController() {
+        let coordinator = createCoordinator(initialScreen: .chatTranscript)
+        let viewController = coordinator.start() as? ChatViewController
+
+        XCTAssertNotNil(viewController)
+    }
+
+    // Delegate
+    func test_backTapped() {
+        coordinator.delegate = { event in
+            switch event {
+            case .backTapped: XCTAssertTrue(true)
+            default: XCTFail()
+            }
+        }
+
+        coordinator.viewModel?.delegate?(.backTapped)
+    }
+
+    func test_closeTapped() {
+        coordinator.delegate = { event in
+            switch event {
+            case .closeTapped(let presentation):
+                XCTAssertEqual(presentation, .doNotPresentSurvey)
+            default: XCTFail()
+            }
+        }
+
+        coordinator.viewModel?.delegate?(.backTapped)
+    }
+
+    func test_renderProps() {
+        let welcomeViewController = coordinator.start() as! SecureConversations.WelcomeViewController
+        coordinator.viewModel?.delegate?(.renderProps(.mock))
+
+        XCTAssertEqual(.mock, welcomeViewController.props)
+    }
+
+    func test_confirmationScreenRequested() {
+        _ = coordinator.start()
+        coordinator.viewModel?.delegate?(.confirmationScreenRequested)
+        let confirmationViewController = navigationPresenter.viewControllers
+            .last as? SecureConversations.ConfirmationViewController
+
+        XCTAssertNotNil(confirmationViewController)
+    }
+
+    func test_mediaPickerRequested() {
+        let welcomeViewController = coordinator.start() as! SecureConversations.WelcomeViewController
```

Comment:
It is possible to avoid force casting, by using  `try XCTUnwrap(coordinator.start() as?SecureConversations.WelcomeViewController)` and marking `test_mediaPickerRequested` method throwing.

Reply:
Fix bfce3ead403f998eef6bcf9d8de660a98d89ce91